pkg/driver/docker/docker.go

```diff
@@ -363,6 +366,191 @@ func (d *dockerDriver) RunDockerDevContainer(
+		d.Log.Debugf("Running docker command: %s %s", d.Docker.DockerCommand, strings.Join(args, " "))
+		err = d.Docker.Run(ctx, args, nil, writer, writer)
+		if err != nil {
+			return err
+		}
+
+		args = []string{"cp", fmt.Sprintf("%s:/etc/group", container.ID), containerGroupFileIn.Name()}
+		d.Log.Debugf("Running docker command: %s %s", d.Docker.DockerCommand, strings.Join(args, " "))
+		err = d.Docker.Run(ctx, args, nil, writer, writer)
+		if err != nil {
+			return err
+		}
+
+		containerPasswdFileIn, err = os.Open(containerPasswdFileIn.Name())
+		if err != nil {
+			return err
+		}
+		defer containerPasswdFileIn.Close()
+		// Update /etc/passwd and /etc/group with the new user UID and GID
+		scanner := bufio.NewScanner(containerPasswdFileIn)
+		containerUid := ""
+		containerGid := ""
+		containerHome := ""
+
+		re := regexp.MustCompile(fmt.Sprintf(`^%s:(?P<password>x?):(?P<uid>.*):(?P<gid>.*):(?P<gcos>.*):(?P<home>.*):(?P<shell>.*)$`, containerUser))
+		for scanner.Scan() {
+			match := re.FindStringSubmatch(scanner.Text())
+			if match == nil {
+				_, err := containerPasswdFileOut.WriteString(fmt.Sprintf("%s\n", scanner.Text()))
+				if err != nil {
+					return err
+				}
+				continue
+			}
+			result := make(map[string]string)
+			for i, name := range re.SubexpNames() {
+				if i != 0 && name != "" {
+					result[name] = match[i]
+				}
+			}
+			containerUid = result["uid"]
+			containerGid = result["gid"]
+			containerHome = result["home"]
+
+			_, err := containerPasswdFileOut.WriteString(fmt.Sprintf("%s:%s:%s:%s:%s:%s:%s\n", containerUser, result["password"], localUid, localGid, result["gcos"], result["home"], result["shell"]))
+			if err != nil {
+				return err
+			}
+		}
+
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+
+		if localUid == "0" || containerUid == "0" || (localUid == containerUid && localGid == containerGid) {
+			return nil
+		}
+
+		containerGroupFileIn, err = os.Open(containerGroupFileIn.Name())
+		if err != nil {
+			return err
+		}
+		defer containerGroupFileIn.Close()
+
+		scanner = bufio.NewScanner(containerGroupFileIn)
+
+		re = regexp.MustCompile(fmt.Sprintf(`^(?P<group>.*):(?P<password>x?):%s:(?P<group_list>.*)$`, containerGid))
+		for scanner.Scan() {
+			match := re.FindStringSubmatch(scanner.Text())
+			if match == nil {
+				_, err := containerGroupFileOut.WriteString(fmt.Sprintf("%s\n", scanner.Text()))
+				if err != nil {
+					return err
+				}
+				continue
+			}
+			result := make(map[string]string)
+			for i, name := range re.SubexpNames() {
+				if i != 0 && name != "" {
+					result[name] = match[i]
+				}
+			}
+
+			_, err := containerGroupFileOut.WriteString(fmt.Sprintf("%s:%s:%s:%s\n", result["group"], result["password"], localGid, result["group_list"]))
+			if err != nil {
+				return err
+			}
+		}
+
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+
+		d.Log.Debugf("Updating container user uid")
+
+		// Copy /etc/passwd and /etc/group back to the container
+		{
```

Comment:
also don't think we need these anonymous blocks

Reply:
corrected